### PR TITLE
Add uuid to staging folder to fix rename issues

### PR DESCRIPTION
**Pull request by Flanker32**
Add uuid to staging folder name, for old artifacts may stay in staging folder which may block user update their artifacts.

### CODE REVIEW

File: azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/handlers/artifact/ArtifactHandlerImplV2.java [hunk 1]
```diff
@@ -192,26 +205,51 @@ private boolean isJavaSERuntime() {
...
+        try {
+            log.info(String.format(RENAMING_MESSAGE, artifact.getAbsolutePath(), DEFAULT_APP_SERVICE_JAR_NAME));
+            FileUtils.rename(artifact, renamedArtifact);
+        } catch (IOException e) {
+            throw new MojoExecutionException(String.format(RENAMING_FAILED_MESSAGE, DEFAULT_APP_SERVICE_JAR_NAME));
+        }
+    }
+
+
+    private boolean isWebConfigExists(final List<File> artifacts) {
```
Comment (andxu) [inline azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/handlers/artifact/ArtifactHandlerImplV2.java]: existsWebConfig

File: azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/handlers/artifact/ArtifactHandlerImplV2.java [hunk 2]
```diff
@@ -31,7 +37,14 @@
     private static final int MAX_RETRY_TIMES = 3;
     private static final String ALWAYS_DEPLOY_PROPERTY = "alwaysDeploy";
 
-    public static final String RENAMING_MESSAGE = "Renaming %s to %s";
+    private static final String WEB_CONFIG = "web.config";
+    private static final String RENAMING_MESSAGE = "Renaming %s to %s";
+    private static final String RENAMING_FAILED_MESSAGE = "Failed to rename artifact to %s, which is required in Java SE environment, " +
+            "please refer https://docs.microsoft.com/en-us/azure/app-service/containers/configure-language-java#set-java-runtime-options for more information.";
```
Comment (andxu) [inline azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/handlers/artifact/ArtifactHandlerImplV2.java]: refer to xxx for details.

File: azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/handlers/artifact/ArtifactHandlerImplV2.java [hunk 3]
```diff
@@ -31,7 +37,14 @@
...
+    public static final String NO_EXECUTABLE_JAR = "No executable jar found in <resources>, please check the configuration";
```
Comment (andxu) [inline azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/handlers/artifact/ArtifactHandlerImplV2.java]: No executable jars are found in target folder according to resource filter %s, please make sure the resource filter is correct and you have built the jar.

File: azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/handlers/artifact/ArtifactHandlerImplV2.java [hunk 4]
```diff
@@ -31,7 +37,14 @@
...
+    public static final String MULTI_FINAL_NAME_JARS = "Multi executable jars with final name %s found in <resources>, please check the configuration";
```
Comment (andxu) [inline azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/handlers/artifact/ArtifactHandlerImplV2.java]: How can this happen? there should be only one final jar built during package.

Comment (Flanker32) [reply azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/handlers/artifact/ArtifactHandlerImplV2.java]: I want to scan artifacts in sub folders of targer folder but app service will not detect them, will fixed it.

File: azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/handlers/artifact/ArtifactHandlerImplV2.java (hunk 3 above)
Comment (Flanker32) [reply azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/handlers/artifact/ArtifactHandlerImplV2.java]: Shall we show `<resources>` information in console with xml format? Or just shows "No executable jar found in target folder according to resource filters specified in <resources>, please make sure the resource filter is correct and you have built the jar."

File: azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/handlers/artifact/ArtifactHandlerImplV2.java (hunk 2 above)
Comment (Flanker32) [reply azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/handlers/artifact/ArtifactHandlerImplV2.java]: Fixed

File: azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/handlers/artifact/ArtifactHandlerImplV2.java (hunk 1 above)
Comment (Flanker32) [reply azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/handlers/artifact/ArtifactHandlerImplV2.java]: Done.

File: azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/handlers/artifact/ArtifactHandlerImplV2.java [hunk 5]
```diff
@@ -192,26 +204,49 @@ private boolean isJavaSERuntime() {
...
+        try {
+            log.info(String.format(RENAMING_MESSAGE, artifact.getAbsolutePath(), DEFAULT_APP_SERVICE_JAR_NAME));
+            FileUtils.rename(artifact, renamedArtifact);
+        } catch (IOException e) {
+            throw new MojoExecutionException(String.format(RENAMING_FAILED_MESSAGE, DEFAULT_APP_SERVICE_JAR_NAME));
+        }
+    }
+
+
+    private boolean existsWebConfig(final List<File> artifacts) {
```
Comment (andxu) [inline azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/handlers/artifact/ArtifactHandlerImplV2.java]: static

File: azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/handlers/artifact/ArtifactHandlerImplV2.java [hunk 6]
```diff
@@ -192,26 +204,49 @@ private boolean isJavaSERuntime() {
...
+                .filter(file->StringUtils.equals(fileName, file.getName())).findFirst().orElse(null);
+        if (executableArtifacts.size() == 0) {
+            throw new MojoExecutionException(NO_EXECUTABLE_JAR);
+        } else if (finalArtifact == null && executableArtifacts.size() > 1) {
+            throw new MojoExecutionException(MULTI_EXECUTABLE_JARS);
+        }
+        return finalArtifact == null ? executableArtifacts.get(0) : finalArtifact;
+    }
+
+    public static boolean isExecutableJar(File file) {
```
Comment (andxu) [inline azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/handlers/artifact/ArtifactHandlerImplV2.java]: private?

File: azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/handlers/artifact/ArtifactHandlerImplV2.java (hunk 5 above)
Comment (Flanker32) [reply azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/handlers/artifact/ArtifactHandlerImplV2.java]: Done

File: azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/handlers/artifact/ArtifactHandlerImplV2.java (hunk 6 above)
Comment (Flanker32) [reply azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/handlers/artifact/ArtifactHandlerImplV2.java]: Done

File: azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/AbstractAppServiceMojo.java [hunk 7]
```diff
@@ -117,10 +121,17 @@ public DeploymentType getDeploymentType() throws MojoExecutionException {
     }
 
     public String getDeploymentStagingDirectoryPath() {
-        final String outputFolder = this.getPluginName().replaceAll(MAVEN_PLUGIN_POSTFIX, "");
-        return Paths.get(
-            this.getBuildDirectoryAbsolutePath(),
-            outputFolder, this.getAppName()
-        ).toString();
+        if (StringUtils.isEmpty(stagingDirectoryPath)) {
```
Comment (andxu) [inline azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/AbstractAppServiceMojo.java]: Shall we delete this folder later?

File: azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/handlers/artifact/ArtifactHandlerImplV2.java (hunk 3 above)
Comment (andxu) [reply azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/handlers/artifact/ArtifactHandlerImplV2.java]: Print user's value in xml or default value for resource filter.

Comment (Flanker32) [reply azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/handlers/artifact/ArtifactHandlerImplV2.java]: Print it with Resource.toString() method, as ArtifactHandlerImplV2 can't access the origin pom.xml , here is an example
`
No executable jar found in target folder according to resource filter 'Resource {targetPath: D:\Personal\WebApp\spring-petclinic\target\azure-webapp\spring-petclinic-webconfig-ab78e30d-b3c4-4e27-9f32-afd6476cfeaa, filtering: false, FileSet {directory: D:\Personal\WebApp\spring-petclinic/target, PatternSet [includes: {*.exec}, excludes: {}]}},Resource {targetPath: D:\Personal\WebApp\spring-petclinic\target\azure-webapp\spring-petclinic-webconfig-ab78e30d-b3c4-4e27-9f32-afd6476cfeaa, filtering: false, FileSet {directory: D:\Personal\WebApp\spring-petclinic/target, PatternSet [includes: {*.war}, excludes: {}]}}', please make sure the resource filter is correct and you have built the jar.
`

File: azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/AbstractAppServiceMojo.java (hunk 7 above)
Comment (Flanker32) [reply azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/AbstractAppServiceMojo.java]: Fixed

**Pull request merged**